src/main/java/org/opensearch/knn/quantization/models/quantizationOutput/BinaryQuantizationOutput.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.models.quantizationOutput;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+/**
+ * The BinaryQuantizationOutput class represents the output of a quantization process in binary format.
+ * It implements the QuantizationOutput interface to handle byte arrays specifically.
+ */
+@NoArgsConstructor
+public class BinaryQuantizationOutput implements QuantizationOutput<byte[]> {
+    @Getter
+    private byte[] quantizedVector;
+
+    /**
+     * Updates the quantized vector with a new byte array.
+     *
+     * @param newQuantizedVector the new quantized vector represented as a byte array.
+     */
+    public void updateQuantizedVector(final byte[] newQuantizedVector) {
```

Comment:
wondering what could be a better way to doing this and more importantly how clients are supposed to use this, how clients will know what length of byte vector in this case needs to provided

Reply:
Yeah I made update to this , with the new one , we are even keeping the same refrence for quantized vector.